### DISCUSSION TITLE
how to update xbvr (3.2.3 -> 3.2.4) in docker container without loosing config and matched videos

### DISCUSSION THREAD

**Discussion by ebertomail** [xbapps/xbvr#769] [General]
Hi,

Due to lack of experience, I wonder how one updates xbvr in a docker container without loosing configurations, movie DB, matched videos, etc.

Cheers!

**Comment by toshski** [comment]
I am not expert in this area and may have it wrong, but this is my understanding.  

I just updated my docker repository setting from pointing to _ghcr.io/xbapps/xbvr:0.3.23_ to _ghcr.io/xbapps/xbvr:latest_ (I could have used ghcr.io/xbapps/xbvr:0.3.24) and that was all I had to do.  Using **latest** just means next time I don't need to do anything, it should find the latest version automatically.  I didn't need to do anything to protect my config (although a backup is always a good idea).  

Your docker setup should have mapping for the /videos and /root/.config/  paths and these would map to somewhere outside your docker image.  The docker run command should have arguments to do this. What you need should be stored under /root/.config/. Assuming this is mapped outside of your docker image, when you update the docker, it's image is rebuilt but wherever your .config folder points will be safe because it is not actually inside the docker.

Most of the details you are talking about in terms of your config, matched videos, etc., are stored in the database which is in /root/.config/xbvr/main.db , so given /root/.config is mapped outside of docker, your database should be safely outside of docker

/root/.config also stores other data as well as the database, eg image cache, scene indexes, heatmaps, but it should all be safe in an upgrade as well.  To be safe, as long as you have a backup copy of main.db.  I think pretty much everything else can be rebuilt be xbvr anyway.
